pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -106,22 +100,32 @@ func (hr *horizontalController) performScaling(ctx context.Context, podAutoscale
 		scaleDirection = scaleDown
 	}
 
-	scale.Spec.Replicas = hr.computeScaleAction(currentDesiredReplicas, targetDesiredReplicas, minReplicas, maxReplicas, scaleDirection)
-	scaleResult, err := hr.scaler.update(ctx, gr, scale)
+	replicasWithRules := hr.computeScaleAction(currentDesiredReplicas, targetDesiredReplicas, minReplicas, maxReplicas, scaleDirection)
```

Comment:
I think we need to handle the case where `replicasWithRules == scale.Spec.Replicas` (no action taken)

Reply:
It's handled just before

Reply:
Alright but what if after applying rules it becomes equal to `scale.Spec.Replicas` ?